src/rules/noMagicNumbersRule.ts

```diff
@@ -68,25 +68,21 @@ export class Rule extends Lint.Rules.AbstractRule {
 
 class NoMagicNumbersWalker extends Lint.RuleWalker {
     // lookup object for allowed magic numbers
-    private allowed: { [prop: string]: boolean } = {};
+    private allowed: Set<string>;
     constructor (sourceFile: ts.SourceFile, options: IOptions) {
         super(sourceFile, options);
 
         const configOptions = this.getOptions();
         const allowedNumbers: number[] = configOptions.length > 0 ? configOptions : Rule.DEFAULT_ALLOWED;
-
-        allowedNumbers.forEach((value) => {
-            this.allowed[value] = true;
-        });
+        this.allowed = new Set(allowedNumbers.map(String));
     }
 
     public visitNode(node: ts.Node) {
         const isUnary = this.isUnaryNumericExpression(node);
-        if (node.kind === ts.SyntaxKind.NumericLiteral && node.parent !== undefined && !Rule.ALLOWED_NODES[node.parent.kind] || isUnary) {
-            const text = node.getText();
-            if (!this.allowed[text]) {
-                this.addFailureAtNode(node, Rule.FAILURE_STRING);
-            }
+        const isNumber = node.kind === ts.SyntaxKind.NumericLiteral && !Rule.ALLOWED_NODES.has(node.parent!.kind);
+        const isMagicNumber = (isNumber || isUnary) && !this.allowed.has(node.getText());
```

Comment:
Seems like the `isUnary` case should also check for `ALLOWED_NODES`, but this is what it did before.